packages/components/pie-modal/src/index.ts

```diff
@@ -216,6 +223,30 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
         `;
     }
 
+    /**
+     * Render leadingAction button depending on prop availability.
+     *
+     * 1. If the prop `leadingAction` is not provided, the button is not rendered.
+     * 2. If the prop `leadingAction` is provided but any of the optional properties
+     * are not provided, they fall back to their default values.
+     *
+     * @private
+     */
+    private renderLeadingAction (): TemplateResult {
+        const { text, variant = 'primary', ariaLabel } = this.leadingAction;
+
+        return html`
+            <pie-button
+                  variant="${variant}"
+                  aria-label="${ariaLabel || nothing}"
```

Comment:
Does this narrate correctly when its applied to the host?

Reply:
Do you mean via screen reader? I did test it on voice over and it was working fine.